build/utils/yarn.go

```diff
@@ -108,9 +118,71 @@ func GetYarnDependencies(executablePath, srcPath string, packageInfo *PackageInf
 		if responseStr == "" {
 			return
 		}
-		err = nil
 	}
 
+	if isV2AndAbove {
+		dependenciesMap, root, err = buildYarnV2DependencyMap(packageInfo, responseStr)
+	} else {
+		dependenciesMap, root, err = buildYarnV1DependencyMap(packageInfo, responseStr)
+	}
+	return
+}
+
+// getVersion gets the current project's yarn version
+func getVersion(executablePath, srcPath string) (string, error) {
+	command := exec.Command(executablePath, "--version")
+	command.Dir = srcPath
+	outBuffer := bytes.NewBuffer([]byte{})
+	command.Stdout = outBuffer
+	errBuffer := bytes.NewBuffer([]byte{})
+	command.Stderr = errBuffer
+	err := command.Run()
+	if _, ok := err.(*exec.ExitError); ok {
+		err = errors.New(err.Error())
+	}
+	return strings.TrimSpace(outBuffer.String()), err
+}
+
+// buildYarnV1DependencyMap builds a map of dependencies for Yarn versions < 2.0.0
+// Pay attention that in Yarn < 2.0.0 the project itself with its direct dependencies is not presented when running the
+// command 'yarn list' therefore the root is built manually.
+func buildYarnV1DependencyMap(packageInfo *PackageInfo, responseStr string) (dependenciesMap map[string]*YarnDependency, root *YarnDependency, err error) {
+	dependenciesMap = make(map[string]*YarnDependency)
+	var depTree Yarn1Data
+	err = json.Unmarshal([]byte(responseStr), &depTree)
+	if err != nil {
+		return
+	}
+
+	locatorsMap := make(map[string]string)
+	for _, curDependency := range depTree.Data.DepTree {
```

Comment:
let's validate that deptree.data is no nil because if it is, this line will throw an ugly panic

Reply:
since depTree.Data is not a pointer, and It is declared above (therefore I can access those structs fields even before the unmarshal) there is no need to check for nil value.
But I added this check for depTree.Data.DepTree